posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,186 @@
+blog_description: "A high level explanation about how to build a devops pipeline with Jenkins as the core controller component "
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Continuous integration and delivery (CI/CD) of Jakarta EE & MicroProfile applications with Open Liberty to Kubernetes by using Jenkins
+Shamjith Antholi https://github.com/shamjithantholi
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+[#Intro]
+== DevOps, CI/CD, Jenkins & OpenLiberty ( to deploy Jakarta EE & MicroProfile application)
+DevOps is a culture or a process, which enables the integration of the software development (dev) and IT operations (ops). It can facilitate the setup of deployment pipeline and helps to improve the speed of software delivery. Automation of the entire build, test, and deployment of the software applications can be achieved through DevOps process. There are tons of efficient applications and process methodologies available in the market that can be readily integrated to the DevOps pipeline, which can help to achieve the wanted results.
+
+CI/CD are acronym’s for continuous integration, and continuous delivery or deployment used in DevOps. Continuous integration is a process where developers frequently merge code changes into a source code repository, which automatically start code build, security scan, and test execution and deployment. Automation of this process enables organizations to release on a more frequent basis without compromising on quality. There are various tools available to automate the CI/CD process. In this blog post we'll discuss how you can use Jenkins to automate the CI/CD process for building, testing, and deploying your Jakarta EE & MicroProfile applications 
+
+Jenkins is an open source automation server, which helps to automate application code build, test, and deployment. Jenkins comes with a hundred’s of plug-ins, which eases the various DevOps pipeline tool integrations into the DevOps pipeline.
+
+Open Liberty is a lightweight Java runtime for building cloud-native applications and microservices. In this blog, it details how you can use Jenkins in CI/CD pipeline for Jakarta EE & MicroProfile application build/test/deployment (to deploy using Openliberty ) . To try Openliberty link:https://openliberty.io[OpenLiberty] 
+
+The following diagram shows a simple architecture for a pipeline that builds and deploys a Jakarta EE & MicroProfile application in a single Kubernetes environment. When a code change is detected by Jenkins, Jenkins builds the code, runs security and static analysis scans, and runs unit tests against the built code. Jenkins then generates a Docker image of the application with its Open Liberty server and saves the image to a Docker repository, Jenkins then trigger the deployment using CLI commands and deploy the Docker image to Kubernetes..
+
+image::/img/blog/liberty-devops-generic-architecture-2.png[Liberty devops generic architecture diagram ,width=70%,align="center"]
```

Comment:
Nice diagram. I can't remember where we got to with checking if we're allowed to use the logos etc. I assume it's okay with @mbroz2 @kinueng ?

Reply:
I created that diagram. As I know, we can use the logos without any modification in articles about the use of same technology ( verified a few from IBM Colleagues as well )

Reply:
This depends on both the logo/company and how the logo is used (there's a difference between talking about a company in a blog post vs implying a relationship with said company). You have to follow the rules of that company to make sure you're using their logos in a way that complies with their rules. For example, Docker's copyright rules are here: https://www.docker.com/legal/trademark-guidelines/
Kubernetes is under the Linux foundation: https://www.linuxfoundation.org/trademark-usage/

Otherwise, you'd need to seek permission to use their logo (which we've done in the past).  If unsure, I would recommend getting input from @CassTucker

Reply:
Each company has guidelines on how to use their logos. The picture in question here contains many logos that will have different usage guidelines.  

For an example, SonarQube's Logo and Usage https://www.sonarqube.org/logos/ has a easy `Can do` and `Can NOT do`. If the guidelines are met, then the usage of SolarQube is fine per the page saying "You can use the SonarQube logo freely in compliance with the guidelines below."

My recommendation is that we invest the time to ensure we are not violating any logo usage guidelines